sklearn/cluster/k_means_.py

```diff
@@ -862,14 +869,24 @@ def __init__(self, n_clusters=8, init='k-means++', n_init=10,
 
     def _check_fit_data(self, X):
         """Verify that the number of samples given is larger than k"""
-        X = check_array(X, accept_sparse='csr', dtype=[np.float64, np.float32])
+        if self.algorithm == "full":
+            order = "C"
+        else:
+            order = None
```

Comment:
_kmeans_elkan coerces the data to C order anyway, so this turns out to be redundant and weird.